sdk-extension/opentelemetry-sdk-extension-aws/src/opentelemetry/sdk/extension/aws/trace/propagation/aws_xray_format.py

```diff
@@ -0,0 +1,283 @@
+                    break
+
+                try:
+                    trace_id = self.parse_trace_id(value)
+                except ValueError:
+                    _logger.error(
+                        (
+                            "Invalid TraceId in X-Ray trace header: '%s' with value '%s'. Returning INVALID span context.",
+                            self.TRACE_HEADER_KEY,
+                            trace_header,
+                        )
+                    )
+                    extract_err = True
+            elif key == self.PARENT_ID_KEY:
+                if not self.validate_span_id(value):
+                    _logger.error(
+                        (
+                            "Invalid ParentId in X-Ray trace header: '%s' with value '%s'. Returning INVALID span context.",
+                            self.TRACE_HEADER_KEY,
+                            trace_header,
+                        )
+                    )
+                    extract_err = True
+                    break
+
+                try:
+                    span_id = AwsXRayFormat.parse_span_id(value)
+                except ValueError:
+                    _logger.error(
+                        (
+                            "Invalid TraceId in X-Ray trace header: '%s' with value '%s'. Returning INVALID span context.",
+                            self.TRACE_HEADER_KEY,
+                            trace_header,
+                        )
+                    )
+                    extract_err = True
+            elif key == self.SAMPLED_FLAG_KEY:
+                if not self.validate_sampled_flag(value):
+                    _logger.error(
+                        (
+                            "Invalid Sampling flag in X-Ray trace header: '%s' with value '%s'. Returning INVALID span context.",
+                            self.TRACE_HEADER_KEY,
+                            trace_header,
+                        )
+                    )
+                    extract_err = True
+                    break
+
+                sampled = self.parse_sampled_flag(value)
+
+        return trace_id, span_id, sampled, extract_err
+
+    def validate_trace_id(self, trace_id_str):
+        return (
+            len(trace_id_str) == self.TRACE_ID_LENGTH
+            and trace_id_str.startswith(self.TRACE_ID_VERSION)
+            and trace_id_str[self.TRACE_ID_DELIMITER_INDEX_1]
+            == self.TRACE_ID_DELIMITER
+            and trace_id_str[self.TRACE_ID_DELIMITER_INDEX_2]
+            == self.TRACE_ID_DELIMITER
+        )
+
+    def parse_trace_id(self, trace_id_str):
+        timestamp_subset = trace_id_str[
+            self.TRACE_ID_DELIMITER_INDEX_1
+            + 1 : self.TRACE_ID_DELIMITER_INDEX_2
+        ]
+        unique_id_subset = trace_id_str[
+            self.TRACE_ID_DELIMITER_INDEX_2 + 1 : self.TRACE_ID_LENGTH
+        ]
+        return int(timestamp_subset + unique_id_subset, 16)
+
+    def validate_span_id(self, span_id_str):
```

Comment:
consider a protected method. Also I like this one-liner: it's a good way to express the intent of the logic. +1 to self-documenting code!